src/settings.js

```diff
@@ -58,10 +58,16 @@ var settings = function() {
   }
 
   this.setLanguage = function(language) {
-    if (language && language.length !== 2) {
-      throw "use 2-char language code (e.g. \"en\")";
+    var languageIntlLocale;
+    try {
+      languageIntlLocale = new Intl.Locale(language).language;
+    } catch (e) {
+      throw "use accepted language code provided by ECMAScript Internationalization API (e.g. \"en\", \"en-GB\")";
     }
-    this.settings.lang = language;
+    if (languageIntlLocale && languageIntlLocale.length !== 2) {
```

Comment:
This is a bit confusing. Control is done with length 2 but error message says 2/4 char language code.

Reply:
SearchApi only accepts 2 char language code as a parameter's value.
Js-client allows customers to input either 2 or 4 char language code, then transform it to 2 char code.
This check is just an extra step to verify if Internationalization API returns the correct format that SearchAPI accepts.